lms/djangoapps/teams/views.py

```diff
@@ -877,6 +877,14 @@ def post(self, request):
         except User.DoesNotExist:
             return Response(status=status.HTTP_404_NOT_FOUND)
 
+        course_module = modulestore().get_course(team.course_id)
+        if ('max_team_size' in course_module.teams_configuration and
+            team.users.count() >= course_module.teams_configuration['max_team_size']):
+            return Response(
+                build_api_error(ugettext_noop("The team is already full.")),
+                status=status.HTTP_400_BAD_REQUEST
```

Comment:
Style-wise, I think using the 400 makes the most sense. It's a bad request in that it's an invalid request to ask to add another person to a team that's already 'full'.